frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx

```diff
@@ -154,7 +154,7 @@ const useColorSchemeFromHash = ({
 
 export const ThemeProvider = (props: ThemeProviderProps) => {
   const schemeFromHash = useColorSchemeFromHash({
-    enabled: getIsEmbeddingIframe(),
+    enabled: isStaticEmbedding() || isPublicEmbed(),
```

Comment:
The reason is that public link and even static embedding could be visited directly outside an iframe. When we do that, the `theme` hash parameter is ignored. And this fixes that bug.